src/index.ts

```diff
@@ -115,12 +119,24 @@ bot.on("messageReactionAdd", async (reaction, user) => {
   handleReaction(reaction, user as User);
 });
 
-bot.on("message", msg => {
+bot.on("message", async msg => {
   if (msg.partial) {
-    return;
+    try {
+      await msg.fetch();
+    } catch (e) {
+      console.log("Something went wrong when fetching the message: ", e);
+    }
+  }
+
+  if (msg.author?.partial) {
+    try {
+      await msg.author?.fetch();
+    } catch (e) {
+      console.log("Something went wrong when fetching the message: ", e);
+    }
```

Comment:
Wouldn't it be better to have one catch block here? If `msg.partial` is falsy I doubt `msg.author.partial` would be truthy anyway

Reply:
I studied a bit more about partials and in this `"message"` event it is not needed at all since the message is only received when the bot is running so we already have all the data in the event.